trino/client/client.go

```diff
@@ -0,0 +1,260 @@
+package client
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trinodb/trino-go-client/trino"
+)
+
+var (
+	boolTrue  bool = true
+	boolFalse bool = false //nolint:unused
+)
+
+// ConnectionConfig constructs a base ServerURI for a Trino connection.
+// if enforces HTTPS encryption by default, but can be configured to use HTTP.
+// user and passerword are optional but required encyption is enabled.
+// ConnectionConfigs are used to provide a connection string to New to create a new Client.
+// It is a convenience struct to make conditional connection string configuration easier.
+/*
+
+Example usage:
+
+conectionConfig := NewConnectionConfig(WithServerHost("trino.example.com:443"),WithUser("read-only-user"), WithPassword("password"))
+
+serverURI, err := connectionConfig.Parse()
+if err != nil {
+	log.Fatal(err)
+}
+
+client, err := New(serverURI)
+if err != nil {
+	log.Fatal(err)
+}
+
+cursor, err := client.Query(context.Background(), "SELECT * FROM my_table WHERE id = ?", 1)
+*/
+
+type ConnectionConfig struct {
+	// Encrypted indicates whether the connection should use HTTPS, optional, default: true
+	Encrypted *bool `json:"encrypted,omitempty"`
```

Comment:
nit: It doesn't seem like having this a pointer adds functionality, but makes the checks more complicated